### PR TITLE
#2641 Full temperature syncing

### LINKED ISSUE #2641: Temperature sync: full sync
## Is your feature request related to a problem? Please describe.

Related issue #2600 

To do all use-cases of the temperature syncing in a single issue is a bit much. This issue will be the code for the actions to scan for sensors

## Describe the solution you'd like

- A `sync` action
- An `updateSyncProgress` action
- A `startResetLogFrequency` action
- A `completeResetLogFrequency` action
- A `errorResettingLogFreqeuncy` action
- A `resetLogFrequency` action - composition of above
- A `startResetAdvertisementFrequency` action
- A `completeResetAdvertisementFrequency` action
- A `errorResetAdvertisementFrequency` action
- A `resetAdvertisementFrequency` action - composition of above
- A `completeSync` action

## Implementation

- As above

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
#### NOTE: Branched from #2640 

Fixes #2641 

## Change summary

- Adds the final actions for temperature syncing
- `syncTemperatures` is the main action which calls all the others.
- resetting actions are all pretty much duplicates
- `createTemperatureLogs` is the main action which converts the created `SensorLog` records to `TemperatureLog`s and also creates the breaches.
- The reducer state is still quite basic and not too useful. Will be easier to iterate on the exact state when making the temperature syncing modal to know exactly what will be needed

## Testing

N/A

### Related areas to think about

There is possibilities for some exceptions to be thrown - going to move forward and see if I can find them all (there are a few going to come up with the app being put into the background etc etc when the scheduling is being run n stuff)

### CODE REVIEW

File: src/actions/TemperatureSyncActions.js [hunk 1]
```diff
@@ -85,9 +103,145 @@ const downloadLogs = sensor => async dispatch => {
...
+  const sensorLogsToGroup = sensorLogs.sorted('timestamp').slice(0, iterateToo);
+  const groupedSensorLogs = chunk(sensorLogsToGroup, 6);
+
+  dispatch(startSavingTemperatureLogs(groupedSensorLogs.length));
+
+  UIDatabase.write(() => {
+    groupedSensorLogs.forEach(sensorLogGroup => {
+      const { isInBreach, mostRecentTemperatureBreach } = sensor;
+      const temperature = Math.min(...sensorLogGroup);
+      const timestamp = Math.min(...sensorLogGroup);
```
Comment (joshxg) [inline src/actions/TemperatureSyncActions.js:177]: These two lines need to have `sensorLogGroup.map(({timestamp}) => timestamp)` - fixed in #2650

File: src/actions/TemperatureSyncActions.js [hunk 2]
```diff
@@ -85,9 +103,145 @@ const downloadLogs = sensor => async dispatch => {
...
+
   return Promise.resolve();
 };
 
+const errorNoSensors = () => ({ type: TEMPERATURE_SYNC_ACTIONS.ERROR_NO_SENSORS });
+const startSync = () => ({ type: TEMPERATURE_SYNC_ACTIONS.START_SYNC });
+const completeSync = () => ({ type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_SYNC });
+
+const syncTemperatures = () => async (dispatch, getState) => {
+  const { isSyncing } = getState();
```
Comment (joshxg) [inline src/actions/TemperatureSyncActions.js:215]: this line should be `const { temperatureSync } = getState()` then `const {isSyncing} = temperatureSync` -- fixed in #2650

File: src/actions/TemperatureSyncActions.js [hunk 3]
```diff
@@ -85,9 +103,145 @@ const downloadLogs = sensor => async dispatch => {
...
+const completeSavingTemperatureLogs = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_SAVING_TEMPERATURE_LOGS,
+});
+
+const createTemperatureLogs = sensor => dispatch => {
+  const { sensorLogs, location } = sensor;
+
+  // Only create TemperatureLogs for the greatest multiple of 6 SensorLogs,
+  // as each SensorLog is a 5 minute log, and each Temperature log a 30 minute log.
+  const iterateToo = sensorLogs.length - (sensorLogs.length % 6);
```
Comment (mark-prins) [inline src/actions/TemperatureSyncActions.js:167]:
```suggestion
  const iterateTo = sensorLogs.length - (sensorLogs.length % 6);
```

Comment (joshxg) [reply src/actions/TemperatureSyncActions.js]: embares

ccdcd42

**Pull request merged**